gcc/rust/privacy/rust-reachability.h

```diff
@@ -46,6 +46,19 @@ class ReachabilityVisitor : public HIR::HIRVisItemVisitor
     : current_level (ReachLevel::Reachable), ctx (ctx), ty_ctx (ty_ctx)
   {}
 
+  /**
+   * Visit all the predicates of all the generic types of a given item, marking
+   * them as reachable or not.
+   */
+  void visit_generic_predicates (
+    const std::vector<std::unique_ptr<HIR::GenericParam>> &generics,
+    ReachLevel item_reach);
+
+  /**
+   * Get the initial reach level for an item based on its visibility.
+   */
+  ReachLevel initial_reach (const HIR::Visibility &item_visibility);
```

Comment:
this could be a static function since you don't need access to self here

Reply:
Oops - this is a typo actually. We should be returning `current_level` instead of `ReachLevel::Reachable`